src/dsp_tools/utils/xmlupload/resource_create_client.py

```diff
@@ -37,18 +38,21 @@ def create_resource(
     ) -> tuple[str, str]:
         """Creates a resource on the DSP server."""
         logger.info(f"Attempting to create resource {resource.id} (label: {resource.label}, iri: {resource.iri})...")
-        resource_json_ld = self._make_json_ld_resource(resource, bitstream_information)
-        res = try_network_action(self.con.post, route="/v2/resources", jsondata=resource_json_ld)
-        iri = res["@id"]
-        label = res["rdfs:label"]
-        return iri, label
-
-    def _make_json_ld_resource(
-        self,
-        resource: XMLResource,
-        bitstream_information: BitstreamInfo | None,
-    ) -> str:
-        return json.dumps(self._make_resource_with_values(resource, bitstream_information), ensure_ascii=False)
+        resource_dict = self._make_resource_with_values(resource, bitstream_information)
+        resource_json_ld = json.dumps(resource_dict, ensure_ascii=False)
+        try:
+            res = try_network_action(self.con.post, route="/v2/resources", jsondata=resource_json_ld)
+            iri = res["@id"]
+            label = res["rdfs:label"]
+            return iri, label
+        except BaseError as e:
+            msg = f"Could not create resource {resource.id} (label: {resource.label}, iri: {resource.iri})"
+            logger.exception(msg)
+            raise BaseError(msg) from e
```

Comment:
The current strategy in this codebase is to write the entire stack trace into the logging file (with `exc_info=True`), and then to omit the ugly stack trace from the error, so that a cleaner stack trace is printed:

```suggestion
            logger.exception(msg, exc_info=True)
            raise BaseError(msg) from None
```

But now I see that in `xmlupload.py`, `_create_resource()`, you catch the BaseError, log it again, and print it. So then you shouldn't handle it here at all, I guess...?